Kernel/Thread.cpp

```diff
@@ -1234,6 +1234,13 @@ DispatchSignalResult Thread::dispatch_signal(u8 signal)
     auto signal_trampoline_addr = process.signal_trampoline().get();
     regs.set_ip(signal_trampoline_addr);
 
+#if ARCH(I386) || ARCH(X86_64)
+    // Userspace flags might be invalid for function entry, according to SYSV ABI.
```

Comment:
At least mentioning the matching section of the specification would be nice here, took me a bit to find it myself.